### PR TITLE
virttest.qemu_vm: make parent bus of device configurable

### PR DESCRIPTION
**Pull request by xutian**
Signed-off-by: Xu Tian <xutian@redhat.com>

### CODE REVIEW

**Comment by xutian** [comment]
@jingzhao84  can you help review this PR, thanks

**Comment by jingzhao84** [comment]
LGTM
create correct command with updated cfg 
cfg:
pci_controllers = root_port root_port2 pcie_switch pcie_switch2 dmi2pci_bridge pci_bridge
slot_root_port2 = 2
type_root_port2 = ioh3420
type_pcie_switch2 = x3130
pci_bus_pcie_switch2 = root_port2
nics += " nic5"
nic_model_nic5 = e1000e
nic_pci_bus_nic5 = pcie_switch2 

cmd:
/usr/libexec/qemu-kvm \
    -S  \
    -name 'avocado-vt-vm1'  \
    -sandbox off  \
    -machine q35  \
    -nodefaults  \
    -vga qxl \
    -device ioh3420,id=root_port,bus=pcie.0,addr=02 \
    -device ioh3420,slot=2,id=root_port2,bus=pcie.0,addr=03 \
    -device x3130-upstream,id=pcie_switch,bus=root_port,addr=00 \
    -device x3130-upstream,id=pcie_switch2,bus=root_port2,addr=00 \
    ..................................
   -device xio3130-downstream,bus=pcie_switch2,id=pcie_switch2.1,addr=01,chassis=4 \
   -device e1000e,mac=9a:61:62:63:64:65,id=idHThnEd,netdev=idKskNRM,bus=pcie_switch2.1,addr=00  \

boot case result:
PASS 1-smp_4.4096m.repeat1.test.Host_RHEL.m7.u3.q35.spice.ovmf.qcow2.virtio_scsi.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.boot

**Comment by xutian** [comment]
thanks @jingzhao84 's test results.  Hi @CongLi  can you help second review this pull request.

Thanks,
Xu

File: virttest/qemu_vm.py [hunk 1]
```diff
@@ -1270,33 +1271,32 @@ def add_disable_modern(devices, dev, dev_type):
...
             Get device parent pci bus by dtype
 
             :param devices: DevContainer object
-            :param params: test params
-            :param dtype: device type
-            :param virtio: is it a virtio device (bool type)
+            :param params: test params for the device
+            :param dtype: device type like, 'nic', 'disk',
+                          'vio_rng', 'vio_port' or 'cdrom'
+            :param pcie: is it a pcie device (bool type)
```
Comment (CongLi) [inline virttest/qemu_vm.py]: typo: 
s/is it/ it is

**Comment by CongLi** [comment]
1024 23:02:50 INFO | PASS 1-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.q35.qcow2.virtio_     blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.balloon_check.balloon_evict     _and_enlarge.balloon_base
3220 22:55:40 INFO | PASS 1-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.q35.virtio_rng.rn     g_egd.qcow2.virtio_blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.rng_bat
2065 22:48:54 INFO | PASS 1-smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u3.q35.qcow2.virtio_     blk.up.virtio_net.RHEL.7.3.x86_64.io-github-autotest-qemu.multi_disk.all_drive_format     _types

And code looks good to me.
But could you please update one typo in doc string?

Acked-by: Cong Li <coli@redhat.com>

**Comment by xutian** [comment]
fixed typo, thanks comments from  @CongLi  and @jingzhao84. Merged

**Pull request merged**